app/models/semester.rb

```diff
@@ -23,6 +23,10 @@ def can_be_destroyed?
     ! (student_applications.exists? || projects.exists?)
   end
 
+  def application_exists(applicant)
+    student_applications.where(applicant: applicant).present?
+  end
```

Comment:
Maybe it's better to have a method on the applicant instead of the semester? Arguably it makes more sense to ask an applicant whether he/she's applied for this semester already than to ask a semester whether a particular applicant has applied.

So instead of `@semester.application_exists` probably something like `@applicant.has_applied_for?(semester)`.